src/software/amazon/ion/impl/lite/IonSequenceLite.java

```diff
@@ -459,45 +456,28 @@ public boolean containsAll(final Collection<?> c) {
         public IonValue[] toArray() {
             checkForParentModification();
 
-            if (size < 1) return EMPTY_VALUE_ARRAY;
+            if (size < 1) {
```

Comment:
apply this in `toArray(T[] array)` as well?

Reply:
We can't have a singleton return for `toArray(T[] array)` for `size == 0`. `array` could be different types and the contract is different. We don't return an empty array for `size == 0` we either set `array[size] = null` if `array.length >= size` or do nothing and return the `array`